openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2166,6 +2188,94 @@ The following is a non-normative example of a Credential Response containing a C
 
 <{{examples/credential_response_sd_jwt_vc.txt}}
 
+# Key Attestations {#keyattestation}
+
+A key attestation is an interoperable, verifiable statement that provides evidence of the authenticity and security properties of a key and its storage component. Keys can be stored in various key storage components, which differ in their ability to protect the private key against extraction and duplication, as well as in the methods used for End-User authentication to unlock key operations. These key storage components may be software-based or hardware-based, and can be located on the same device as the Wallet, on external security tokens, or on remote services that enable cryptographic key operations. Key attestations are signed by the Wallet Provider or the Wallet's key storage component itself.
+
+A Wallet MAY provide key attestations to inform the Credential Issuer about the properties of the provided cryptographic public keys, e.g. for proof types sent in the Credential Request. Credential Issuers may want to evaluate these key attestations to determine whether the keys meet their own security requirements, based on the trust framework in use, regulatory requirements, laws, or internal design decisions. An Issuer SHOULD communicate this requirement to evaluate key attestations through its metadata or using some sort of out-of-band mechanism.
+
+There are two flows how key attestations can be used within Credential issuance:
+- The Wallet uses the `jwt` proof type in the Credential Request to create a proof of possession of the key and adds the key attestation in the JOSE header as specified in (#jwt-proof-type).
+- The Wallet uses the `attestation` proof type in the Credential Request with the key attestation without a proof of possession of the key itself as specified in (#attestation-proof-type).
```

Comment:
umm... I don't think this really reflects what we discussed in the WG: https://github.com/openid/OpenID4VCI/pull/389#issuecomment-2386849999

I think key attestation proof type should be adjusted to both options documented in the comment above, to support both key attestation proof type with nonce in key attestation and key attestation proof type with nonce in PoP. and key attestation in jwt proof type being removed.

because if we go the current path. an option to do key attestation without a nonce is only limited to key attestation per Proof, right? because i don't think jwt proof type has been extended to include more than one key.

Reply:
That's incorrect, `jwt` proof type allows multiple keys: https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-8.2-2.4

Reply:
so this was already resolved by tue WG call. my confusion was coming from the fact that it was not clear that adding key attestation to jwt proof type allows to attest for multiple keys.